ui/src/layouts/storage/mod.rs

```diff
@@ -254,23 +235,30 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                     format!("{}", get_local_text("files.storage-current-size")),
                                     span {
                                         class: "count",
-                                       format!("{}", storage_size.read().1),
+                                       format!("{}", storage_controller.read().storage_size.1),
                                     }
                                 },
                             )
                         }
                     }
                 }
+                UploadProgressBar {
+                    are_files_hovering_app: upload_file_controller.are_files_hovering_app,
+                    files_been_uploaded: upload_file_controller.files_been_uploaded,
+                    disable_cancel_upload_button: upload_file_controller.disable_cancel_upload_button,
+                    on_update: move |files_to_upload: Vec<PathBuf>|  {
+                        add_files_in_queue_to_upload(upload_file_controller.files_in_queue_to_upload, files_to_upload, window);
+                    },
+                    on_cancel: move |_| {
+                        let _ = tx_cancel_file_upload.send(true);
+                        let _ = tx_cancel_file_upload.send(false);
```

Comment:
Why is a second boolean being sent to the channel?

Reply:
In this case wouldnt it be better to move the cancel field into the inner portion so it would reset on the next iteration? Otherwise you could probably use `AtomicBool` instead to signal to the loop that its being canceled and then reset it in the next iteration.